### PR TITLE
add stubs for 2 edge cases where need PSY bus types

### PR DESCRIPTION
**Pull request by luke-kiernan**
Two small POM test fixes, needed to make tests pass on POM's MBC/IEC PR. Might be a cleaner way--change POM to conform to IOM instead of vice versa--but this'll do for now.

### CODE REVIEW

File: src/quadratic_approximations/incremental.jl [hunk 1]
```diff
@@ -165,8 +165,12 @@ function _add_generic_incremental_interpolation_constraint!(
...
     # Retrieve all required variables from the optimization container
     # Retrieve original variable for DCVoltage from the Bus
-    # FIXME edge case: DCVoltage. W is InterconnectingConverter but key says DCBus.
-    x_var = get_variable(container, R, W)  # Original variable (domain of function)
+    if R <: DCVoltage
+        # workaround for the fact that we can't write PSY.DCBus.
+        x_var = get_variable(container, R, component_for_hvdc_interpolation(nothing))
+    else
+        x_var = get_variable(container, R, W)  # Original variable (domain of function)
+    end
```
[the remark this answers is not in the document]
Comment (luke-kiernan) [reply src/quadratic_approximations/incremental.jl:173]: Yeah this is how stubs work

File: src/common_models/add_constraint_dual.jl [hunk 2]
```diff
@@ -20,7 +20,8 @@ function add_constraint_dual!(
     model::NetworkModel{T},
 ) where {T <: AbstractPowerModel}
     if !isempty(get_duals(model))
-        devices = get_available_components(model, IS.InfrastructureSystemsComponent, sys)
+        # component is ACBus, but we don't have PSY as a dependency.
+        devices = get_available_components(model, component_for_network_dual(nothing), sys)
```
[the remark this answers is not in the document]
Comment (luke-kiernan) [reply src/common_models/add_constraint_dual.jl:24]: yeah this is how stubs work--call doesn't make sense if POM isn't loaded

File: src/common_models/interfaces.jl [hunk 3]
```diff
@@ -245,3 +245,15 @@ Only called in `emulation_model.jl`: that file's contents and this function shou
...
+Component type associated with network duals. Returns ACBus when passed in `nothing`.
+Only called in a single spot in `add_constraint_dual!` for the network model.
+"""
+function component_for_network_dual end
+
+"""
+Component type associated with hvdc interpolation constraints. Returns DCBus when passed in
+`nothing`. Only called in a single spot in `incremental.jl`.
+"""
+function component_for_hvdc_interpolation end
```
[the remark this answers is not in the document]
Comment (luke-kiernan) [reply src/common_models/interfaces.jl:293]: Similarly silly.

File: test/mocks/mock_system.jl [hunk 4]
```diff
@@ -3,70 +3,39 @@ Minimal mock for PSY.System.
...
-IOM.get_base_power(sys::MockSystem) = sys.data.base_power
-IOM.stores_time_series_in_memory(sys::MockSystem) = sys.data.stores_in_memory
-IOM.stores_time_series_in_memory(data::MockSystemData) = data.stores_in_memory
+InfrastructureOptimizationModels.get_base_power(sys::MockSystem) = sys.base_power
+InfrastructureOptimizationModels.stores_time_series_in_memory(sys::MockSystem) =
+    sys.stores_in_memory
 
 function IOM.get_available_components(::NetworkModel, ::Type{T}, sys::MockSystem) where {T}
     return get_components(T, sys)
 end
```
[the remark this answers is not in the document]
Comment (luke-kiernan) [reply test/mocks/mock_system.jl:35]: okay this seems like something potentially worth addressing...but imo it's a part of a deeper problem, which is that emulation models (and to a lesser degree, decision models) aren't well-integrated or well-tested yet. I'd prefer to do that in a separate PR

File: src/common_models/interfaces.jl [hunk 5]
```diff
@@ -145,6 +145,40 @@ For thermals, equivalent to `get_must_run`, but that implementation belongs in P
...
+
+"Extension point: forecast horizon configured on the system."
+function get_forecast_horizon end
+
+"Extension point: summary table of forecasts on the system."
+function get_forecast_summary_table end
+
+"Extension point: transform single time series into deterministic forecasts on the system."
+function transform_single_time_series! end
+
```
Comment (jd-lara) [inline src/common_models/interfaces.jl:175]: I think several of these will cause name clashes with IS no ? Or are methods that should live in IS

File: src/quadratic_approximations/incremental.jl [hunk 6]
```diff
@@ -165,8 +165,12 @@ function _add_generic_incremental_interpolation_constraint!(
 
     # Retrieve all required variables from the optimization container
     # Retrieve original variable for DCVoltage from the Bus
-    # FIXME edge case: DCVoltage. W is InterconnectingConverter but key says DCBus.
-    x_var = get_variable(container, R, W)  # Original variable (domain of function)
+    if R <: DCVoltage
```
Comment (luke-kiernan) [inline src/quadratic_approximations/incremental.jl]: Remove. Exception should be fixed at the source: associate with the matching component, not DCBus
